demos/dotnet/dotnet/CscAction.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace dotnet
+{
+    internal static class CscAction
+    {
+        public static bool Execute(ProjectProperties properties, Log log)
+        {
+            Console.WriteLine("compiling");
+            var processSettings = new ProcessStartInfo
+            {
+                FileName = properties.CscPath,
+                Arguments = properties.FormatCscArguments()
+            };
+
+            log.WriteLine("Executing {0}", processSettings.FileName);
+            log.WriteLine("Csc Arguments: {0}", processSettings.Arguments);
+
+            processSettings.CreateNoWindow = true;
+            processSettings.RedirectStandardOutput = true;
+            processSettings.UseShellExecute = false;
+
+            Process cscProcess;
+            try
+            {
+                cscProcess = Process.Start(processSettings);
+            }
+            catch (Win32Exception)
+            {
+                Console.WriteLine("ERROR: csc.exe needs to be on the path.");
```

Comment:
If the csc.exe is not found in the properties.CscPath, the CscPath property is default set to 'csc.exe'. If csc.exe is not found anywhere up the path (from the current directory), this error is reported. Where we get csc.exe needs to be fixed as right now it basically just looks for it anywhere within .NET framework and hopes one exists. Once that is fixed, I think, that is when we can address proper error handling. Thoughts?